### ISSUE TITLE
4E4D07D3 - Ridge Racer 6

### ISSUE THREAD

**Issue by benvanik** [xenia-project/game-compatibility#21]
[Xbox Marketplace](http://marketplace.xbox.com/en-US/Product/Ridge-Racer-6/66acd000-77fe-1000-9115-d8024e4d07d3)
[Xbox Marketplace (Demo)](http://marketplace.xbox.com/en-US/Product/Ridge-Racer-6-Demo/00000000-0000-400d-80d0-00004e4d07d3)

v4.0.0.0

Tested against benvanik/xenia@ddbab8216954074a3f93536e7a9fb28ecdca8956.

* [ ] Bad save init - XamShowDeviceSelectorUI/errors
* [ ] Texture flickering (color inversion?)
* [ ] XMA audio in menus progressively plays more garbage (like #18)
* [ ] Post-processing issues in game
* [ ] Cube mapping not supported in shaders
* [ ] Slow audio callbacks during menus, but ok in game

**Comment by Parovozik** [comment]
OpenGL. Boot OK. Menu OK. Ingame FAIL.
Vulkan. Boot OK. Menu OK. Ingame preview track and after crash.
Log (Vulkan)
[xenia_RidgeRacer6_VK.zip](https://github.com/xenia-project/game-compatibility/files/506356/xenia_RidgeRacer6_VK.zip)
![bandicam 2016-10-03 15-23-55-234](https://cloud.githubusercontent.com/assets/19775644/19039720/be4d7380-8989-11e6-8af1-e7716d97506a.jpg)

**Comment by Parovozik** [comment]
Tested on (https://github.com/benvanik/xenia/commit/761ae4eeeecf9a993ca6e0075b6d601b63d7deb8)
# Issues:
Partially broken graphics, random crashes, save-states not works
Performance is good, Sound works
For run ingame needed skip preview track!

# LOGs:
[RidgeRacer6_OpenGL.txt](https://github.com/xenia-project/game-compatibility/files/926422/RidgeRacer6_OGL.txt)
[RidgeRacer6_Vulkan.txt MEGA cloud](https://mega.nz/#!pxxFyK4Z!6q_tTTZLSLOclrQBOUOOr_mg3OCn8irhUoYXBsycIQA)

[YouTube RidgeRacer6 Vulkan](https://www.youtube.com/watch?v=CsQbEPEcIVY)
[YouTube RidgeRacer6 OpenGL](https://www.youtube.com/watch?v=6d55RI8PBGk)

# Screenshots:
Vulkan:
![xenia ridge6 vk mp4_snapshot_00 49_ 2017 04 17_22 05 35](https://cloud.githubusercontent.com/assets/19775644/25100784/2e0a1726-23ba-11e7-8e62-827937e59ae5.jpg)
Vulkan:
![xenia ridge6 vk mp4_snapshot_01 57_ 2017 04 17_22 05 53](https://cloud.githubusercontent.com/assets/19775644/25100788/3606c334-23ba-11e7-9f81-e2d358b7de20.jpg)
OpenGL:
![xenia ridge6 ogl mp4_snapshot_01 08_ 2017 04 17_22 06 16](https://cloud.githubusercontent.com/assets/19775644/25100802/3da9e5ee-23ba-11e7-8425-9807dfb35a02.jpg)
OpenGL:
![xenia ridge6 ogl mp4_snapshot_00 35_ 2017 04 17_22 06 26](https://cloud.githubusercontent.com/assets/19775644/25100817/48001586-23ba-11e7-9fe9-c0335bee9953.jpg)

# Labels:
kernel-save-file-errors
gpu-corrupt-drawing
gpu-missing-drawing
state-gameplay

**Comment by 4XPG** [comment]
![aaaaah](https://user-images.githubusercontent.com/4043537/28416432-2ec90254-6d7e-11e7-992a-4d62b43316e7.png)
Everything's in red instead of blue since yesterday's build (Vulkan)

**Comment by Ufinexa** [comment]
Here is my run through of Ridge Racer 6 in DirectX 12 :) Spec list is in description. [https://youtu.be/ilUzRIu8Dag](url)

**Comment by GrandGTA** [comment]
I think they should focus on fixing save game issues before the graphical bugs because when the game crashes or freezes at some point, at least you still have the save game and you can do it again with no worries at all...

**Comment by GrandGTA** [comment]
Here is some screenshots of my run in Ridge Racer 6 Demo.. Seems to be the graphics are suffering on Low End PCs..

Specs:
Intel Core I7-6700 3.40 GHz (8 CPUs)
NVIDIA Quadro K620
32GB RAM
SSD 94GB

Vulkan Obsolete:

![aa](https://user-images.githubusercontent.com/56748157/84010955-fef01180-a9a7-11ea-9a17-59c77be03fd2.PNG)

![saa](https://user-images.githubusercontent.com/56748157/84010987-07e0e300-a9a8-11ea-842b-bccb1b27cac1.png)

Pros:
Some geometry loads properly
Less lag or sometimes no lag
Playable even with graphic bugs
Working particles
The best option to play with crap GPUs like mine
Less flickering trees

Cons:
Some texture bugs that gives me epilepsy
Shadows and other shaders have bugs

D3D12 (ngl i killed my computer after doing this):

![gsaa](https://user-images.githubusercontent.com/56748157/84011036-18915900-a9a8-11ea-9b37-c9b44adcbada.png)

![jhgfds](https://user-images.githubusercontent.com/56748157/84011053-221ac100-a9a8-11ea-9c71-57f840b0c52b.png)

Pros:
Most textures and shaders are fixed
The car select menu now loads the car models properly

Cons:
The best option to cremate your computer
Only works for High End PCs with the best GPUs
Extreme lag
Have to play on bumper screen for Low End PCs
Car takes too long to load when racing (At my run, the car fully loaded at least 2-3 mins)
Basically unplayable for people with Potato PCs
Actually a high chance to overheat your PC completly

**Comment by GrandGTA** [comment]
I tested the RR6 Demo again but this time, in the Canary Build... I used the same settings from the Master build... Nothing was different from both D3D12 and Vulkan GPUs.. The only difference from the Canary build is that freezing is a new addition ... In the Master build, i restart and restart the race and no freezes occur but in the Canary build, freezing got added making the game practically impossible to complete...

**Comment by GrandGTA** [comment]
I got one question.. What are the requirements to run Ridge Racer 6 in D3D12 without lag?

**Comment by Margen67** [comment]
> 
> 
> I got one question.. What are the requirements to run Ridge Racer 6 in D3D12 without lag?

Questions belong on the [Xenia Discord](https://discord.gg/Q9mxZf9).

**Comment by GrandGTA** [comment]
I just knew that Ridge Racer 7 has most of the elements of Ridge Racer 6, just with new features, and also RR7 can be fully played with no game-breaking bugs on RPCS3 which is also an emulator... Try checking RPCS3 and playing RR7 because you might fix some bugs that break the game and the fact that RR7 is just RR6 ported to the PS3 means there is chance that you might find some fixes...

**Comment by GrandGTA** [comment]
Any progress on saving issues??

**Comment by GrandGTA** [comment]
Excuse me, is there any Save workaround???

**Comment by GrandGTA** [comment]
Will there be Online Support on Ridge Racer 6???

**Comment by ballistic1986** [comment]
[https://www.youtube.com/watch?v=KJsHtPHOEYQ&feature=youtu.be&ab_channel=TheTacticalNukeRemixes](url)

**So Far:**

- Game runs for at least 2 courses with the 3rd one being nearly certain it will crash
- Rendering glitches which can be easily fixed with F5
- Full 60fps at 2x resolution achieved, with lower frames at times where rendering gets glitched (Depends greatly on PC Specifications)
- Saves are still not functional, so no way to progress through the game till its end.
- no DLC support
- Overall rendering of cars and environment is 1:1 with the console

PC Specs:
Intel i9-7960x 2.8ghz
Nvidia Geforce GTX 1080Ti
128GB DDR4 RAM

**Comment by GrandGTA** [comment]
> * Overall rendering of cars and environment is 1:1 with the console

I see no problem with Overall Rendering except for the trees... Trees flickering is the only thing I found which is a rendering problem.. Nothing else..

**Comment by Back2Life888** [comment]
Trees don't flicker in Rave City Riverfront for some odd reason..

**Comment by Back2Life888** [comment]
Hey guys, I will try to finish this game with the 1730 pull request build that has the save fixed.. Wish me luck!

**Comment by Back2Life888** [comment]
> Hey guys, I will try to finish this game with the 1730 pull request build that has the save fixed.. Wish me luck!

Actually, no.. I can't progress any further.. The rest of the courses are just crashes and stuff..

**Comment by Back2Life888** [comment]
I know that asking things to get fixed is not allowed here but will you guys fix the FPS problem?? Like with more than 60 FPS, the game speeds up instead of running in a smooth FPS.. Like a 120 FPS X360 game would be perfect..

**Comment by Back2Life888** [comment]
Hey guys, I know that begging for bug fixes is not allowed but I am just gonna ask if the More than 60 FPS bug will be fixed.. I wanna see this game in 120 FPS but the problem is that when you do play it in 120 FPS, it will speed up the game instead of making it smooth..

**Comment by ballistic1986** [comment]
### Course Bugs (As Of Today)

**Disclaimer: This is purely based on my experiences, I am not aware of bugs on other player's clients nor am I saying it is a universal bug.**

**Status Key**

- Perfect - The course plays with no problems whatsoever, you should be able to complete it without any problems
- Somewhat Playable - The course usually plays perfectly, however there is a small chance the game will freeze
- Mostly Unplayable - The course has a high chance of crashing and is very unlikely the player will finish the race.
- Unplayable - The course will crash, the player will not finish the race as the game will always freeze.

**Forward Courses**

- Lakeshore Drive: The course plays perfectly until the last bend before the finish line at the end of Lap 2, where it permanently freezes, However very rarely does it allow the player to finish the race (Status: Mostly Unplayable)
- Rave City Riverfront: No problems here (Status: Perfect)
- Southbay Docks: Course plays perfectly until the S curve before the bridge ramp where it permanently crashes (Status: Unplayable)
- Island Circle: Course plays perfectly until it has a likely chance the game permanently freezes before the finish line after Lap 1 (Status: Somewhat Playable)
- Surfside Resort: Game permanently freezes most of the time within 5 seconds of the race starting, otherwise the game freezes after Lap 1. (Status: Unplayable)
- Highland Cliffs: Course plays perfectly ONLY if the player drives right next to the right wall on the straight through the finish line, otherwise the game permanently freezes. (Status: Mostly Playable)
- Harborline 765: The game is quite likely to crash at the chicane on Lap 3 otherwise the course will play fine. Using special machines will make the replay crash, however using normal machines will run the replay fine. (Status: Somewhat Playable)
- Aviator Loop: No problems here (Status: Perfect)
- Seacrest District: The course runs fine but crashes just before the finish line on Lap 1 (Status: Unplayable)
- Laketop Parkway: Same issue as Lakeshore Drive
- Midtown Parkway: The game is very likely to crash at the chicane, much like Harborline 765 (Status: Mostly Unplayable)
- Airport Lap: The game always crashes after the orange bridge (Status: Unplayable)
- Sunset Heights: Same issue as Seacrest District
- Crossbay Tunnel: No problems here (Status: Perfect)
- Downtown Rave City: No problems here (Status: Perfect)

**Reverse Courses**

- Lakeshore Drive R: No problems here (Status: Perfect)
- Rave City Riverfront R: No problems here (Status: Perfect)
- Southbay Docks R: Game crashes right after the bridge jump on Lap 1 (Status: Unplayable)
- Island Circle R: Game sometimes crashes on the chicane after the finish line (Status: Somewhat Playable)
- Surfside Resort: Game crashes mostly after the bend before the finish line, if it does not, it will crash sometime after (Status: Unplayable)
- Highland Cliffs R: Same issue as Forward Highland Cliffs, however the player needs to drive right next to the left wall this time across the finish line.
- Harborline 765: Same issue as Island Circle R
- Aviator Loop R: No problems here (Status: Perfect)
- Seacrest District R: Game crashes after the bend after the finish line on Lap 2 (Status: Unplayable)
- Laketop Parkway R: No issues here (Status: Perfect)
- Midtown Parkway R: Game crashes after the narrow alleyway with the shops on Lap 2 (Status: Unplayable)
- Airport Lap R: Game crashes after the finish line on Lap 2 (Status: Unplayable)
- Sunset Heights R: Game crashes after the finish line on Lap 2 (Status: Unplayable)
- Crossbay Tunnel R: No problems here (Status: Perfect)
- Downtown Rave City R: No problems here (Status: Perfect)

In conclusion, it seems that courses within the same area have the same bugs, For example, Harborline 765, Midtown Parkway and Island Circle all crash inside of the chicane. Sunset Heights and Seacrest district crash after the finish line. However all Rave City tracks work perfectly.

Cars do not render correctly however it is immediately fixed with F5.

**RNG Bug**
For some odd reason, the original announcer will say this dialogue at any time "That's a great counter! You're one heck of a driver, you gotta teach me!". This dialogue will freeze the game, right after he says the word "counter". So it is advised you do not select this announcer as an option.

**Comment by Back2Life888** [comment]
All courses crashes.. You should change these status to:

Perfect = Very Rarely Chance to Crash
Somewhat Playable = Crashes 2-6 Times but next time may not crash
Mostly Unplayable = 1 Lap goes fine but the rest is History
Unplayable = Can't even get past 0 Lap (Race Beginning, 1st Lap only starts after you hit the Finish/Start Line)

**Comment by Gliniak** [comment]
@ballistic1986 @Back2Life888 Can you try some of these unplayable maps with setting: ``disable_global_lock = true``

In my case it made ``Airport Lap`` playable. No crashes. only some glitches that can be resolved by pressing F5

**Comment by Back2Life888** [comment]
I am scared at blowing up my computer..

**Comment by ballistic1986** [comment]
> @ballistic1986 @Back2Life888 Can you try some of these unplayable maps with setting: `disable_global_lock = true`
> 
> In my case it made `Airport Lap` playable. No crashes. only some glitches that can be resolved by pressing F5

Nope, did not change anything for me

**Comment by Back2Life888** [comment]
Hey Gliniak, keep messing around with the Xenia settings.. Try every single configuration to find some kind of workaround for this game to be playable to the end..

**Comment by ballistic1986** [comment]
I forgot to mention, before a crash the audio cuts out several seconds before. So it is just a silent drive before an inevitable crash

**Comment by Back2Life888** [comment]
The audio should be the cause of the crash.. If someone fixes the audio, probably this game would be state-playable now.. But that is just my guess.. I dunno what ACTUALLY causes the crashes..

**Comment by Back2Life888** [comment]
I discovered a bug. When using the Assoluto Pronzione and shifting gears up, the engine sound will mute for a split second.

**Comment by ballistic1986** [comment]
> I discovered a bug. When using the Assoluto Pronzione and shifting gears up, the engine sound will mute for a split second.

That is not a bug, that is just how the Pronzione sounds like.

**Comment by Back2Life888** [comment]
![All Crashes](https://user-images.githubusercontent.com/76780043/115872311-3d81a400-a474-11eb-8ab1-d133dac10420.png)
I marked all of the areas that the game could crash. If you drive through these red areas, there is a chance the game would crash.

**Comment by ballistic1986** [comment]
> ![All Crashes](https://user-images.githubusercontent.com/76780043/115872311-3d81a400-a474-11eb-8ab1-d133dac10420.png)
> I marked all of the areas that the game could crash. If you drive through these red areas, there is a chance the game would crash. Rave City courses are not included as they are fully playable.

Rave City is always at night time, i think it's the sun that's causing the crashes (possibly)

**Comment by Back2Life888** [comment]
> > ![All Crashes](https://user-images.githubusercontent.com/76780043/115872311-3d81a400-a474-11eb-8ab1-d133dac10420.png)
> > I marked all of the areas that the game could crash. If you drive through these red areas, there is a chance the game would crash. Rave City courses are not included as they are fully playable.
> 
> Rave City is always at night time, i think it's the sun that's causing the crashes (possibly)

Oh I don't think so.

**Comment by Slo2018** [comment]
![2021-05-02_021547](https://user-images.githubusercontent.com/45156187/116798075-75c85880-aaec-11eb-963f-f172541434ad.png)

How to fix this?

**Comment by Back2Life888** [comment]
> ![2021-05-02_021547](https://user-images.githubusercontent.com/45156187/116798075-75c85880-aaec-11eb-963f-f172541434ad.png)
> 
> How to fix this?

Press F5

**Comment by Back2Life888** [comment]
> > Hey guys, I will try to finish this game with the 1730 pull request build that has the save fixed.. Wish me luck!
> 
> Actually, no.. I can't progress any further.. The rest of the courses are just crashes and stuff..

HEY GUYS WITH THE NEW RTV, SOUTHBAY DOCKS AND SOME OTHER TRACKS ARE NOW PLAYABLE!!

**Comment by Back2Life888** [comment]
And also, the glitching car thing is not happening on me anymore. I don't need to press F5.

**Comment by ballistic1986** [comment]
> > > Hey guys, I will try to finish this game with the 1730 pull request build that has the save fixed.. Wish me luck!
> > 
> > 
> > Actually, no.. I can't progress any further.. The rest of the courses are just crashes and stuff..
> 
> HEY GUYS WITH THE NEW RTV, SOUTHBAY DOCKS AND SOME OTHER TRACKS ARE NOW PLAYABLE!!

It does? Still crashes for me, Il update Xenia.

**Comment by Back2Life888** [comment]
> > > > Hey guys, I will try to finish this game with the 1730 pull request build that has the save fixed.. Wish me luck!
> > > 
> > > 
> > > Actually, no.. I can't progress any further.. The rest of the courses are just crashes and stuff..
> > 
> > 
> > HEY GUYS WITH THE NEW RTV, SOUTHBAY DOCKS AND SOME OTHER TRACKS ARE NOW PLAYABLE!!
> 
> It does? Still crashes for me, Il update Xenia.

yeah but I got lucky.

**Comment by TheTacticalNuke** [comment]
Working replay of Southbay Docks. Audio is broken as usual.

https://www.youtube.com/watch?v=8MEYB_eJ9sc

**Comment by ballistic1986** [comment]
Working replay of Southbay Docks. Game audio has crashed (as usual)

https://www.youtube.com/watch?v=8MEYB_eJ9sc

**Comment by ballistic1986** [comment]
Im happy to say that apart from obvious technical glitches which can be rectified with F4

Ridge Racer 6 is now fully playable!

Thanks to the person who fixed the audio errors, the game no longer crashes. However very small error, the main menu music does not loop so it plays only once. Apart from that everything works perfectly!

**Comment by Back2Life888** [comment]
> Im happy to say that apart from obvious technical glitches which can be rectified with F4
> 
> Ridge Racer 6 is now fully playable!
> 
> Thanks to the person who fixed the audio errors, the game no longer crashes. However very small error, the main menu music does not loop so it plays only once. Apart from that everything works perfectly!

Yeah but that is in Canary. Anything Canary is not considered official yet. Once it gets merged with master, that's where you consider it state-playable. There are still some bugs, glitches and problems like:

-The menu music loop that you mentioned
-Ridge Racer V intro sped up in the giant screen (Downtown Rave City, Harborline 765, Crossbay Tunnel has these jumbotrons. They play past Ridge Racer intros like Rave Racer, R4, and others alike but some are bugged like the RRV intro.)
-Flickering trees like in RPCS3 with their RR7 emulation
--Car mesh messed up (F5 is workaround and not a fix. Plus it's very slow taking about 30-40 seconds on mid-end PCs)
-Eats CPU alot.

**Comment by ballistic1986** [comment]
> > Im happy to say that apart from obvious technical glitches which can be rectified with F4
> > Ridge Racer 6 is now fully playable!
> > Thanks to the person who fixed the audio errors, the game no longer crashes. However very small error, the main menu music does not loop so it plays only once. Apart from that everything works perfectly!
> 
> Yeah but that is in Canary. Anything Canary is not considered official yet. Once it gets merged with master, that's where you consider it state-playable. There are still some bugs, glitches and problems like:
> 
> -The menu music loop that you mentioned -Ridge Racer V intro sped up in the giant screen (Downtown Rave City, Harborline 765, Crossbay Tunnel has these jumbotrons. They play past Ridge Racer intros like Rave Racer, R4, and others alike but some are bugged like the RRV intro.) -Flickering trees like in RPCS3 with their RR7 emulation -Car mesh messed up (Enabling d3d12_clear_memory_page_state will fix this without F5 at the cost of framerate) -Eats CPU alot.

Ah yes, I forgot to mention it was in Canary. Thanks for the follow up.

**Comment by mrwallace888** [comment]
### Xenia version

https://github.com/xenia-project/xenia/commit/6b45cf84472c26436d4a5db61a7b50dab301e398

### Issues

The rendering is completely borked, essentially. When picking a machine on the machine selection screen, vehicles render fine. But only for the first time. Once I get into a race, I can't see literally anything unless I use bumper cam, because the polygons for the machine are essentially all over the place. The tail lights are still there, and the Kamata badge is as well, but everything else has essentially buggered off. Nothing like being visually obstructed by a giant stretched black wall, am I right?

After the race, if I go back to machine selection, the cars are borked there as well. Like I said it renders fine the first time but after a race and going back, it's messed up. White tires, glitched decals, etc.

I am unsure if I have to configure something differently or not to resolve this, as seeing gameplay on YT for other Xenia users, the game seems to work fine for the most part.

Also sometimes, when I race the audio eventually crashes leaving me with silence. Less than a minute later, then the game freezes/crashes.

### Log

[xenia.log.zip](https://github.com/xenia-project/game-compatibility/files/8129253/xenia.log.zip)


### Screenshots

![xenia](https://user-images.githubusercontent.com/39398330/155446553-72f281ed-b140-4017-875f-920a680accf7.png)
![xeniawtf](https://user-images.githubusercontent.com/39398330/155446558-46ac4e25-90cc-45eb-8750-e976a8ed0fc9.png)
![xeniawtf2](https://user-images.githubusercontent.com/39398330/155446561-19ea53d2-9b6c-49aa-a1e9-ac8c49da42e0.png)

**Comment by Back2Life888** [comment]
Bruh.. Press F5 to fix the car rendering. And switch to Xenia Canary to stop the crashes. Back track before posting...

**Comment by Back2Life888** [comment]
And next time you ask for help, go to the Discord Server and don't ask here.

**Comment by Back2Life888** [comment]
DLCs are fixed now. Managed to install the Galaga Crinale.

**Comment by SRLSPEED** [comment]
There are graphics bugs in Vulkan engine operation and Vsync off causes the game to speed up to 2x, etc

**Comment by Margen67** [comment]
Xenia Canary issue: https://github.com/xenia-canary/game-compatibility/issues/156

**Comment by madmax2000** [comment]
In the config clear_memory_page_state = true

Or in the Xenia Manager Clear GPU Cache = true (is the same)

And there are no more visual glitches with the car models (no need for F5 anymore)

The game works perfect than!

**Comment by Back2Life888** [comment]
> In the config clear_memory_page_state = true
> 
> Or in the Xenia Manager Clear GPU Cache = true (is the same)
> 
> And there are no more visual glitches with the car models (no need for F5 anymore)
> 
> The game works perfect than!

Is this in Canary? Cause if it is, it needs to be in its separate Canary issue, the one above your message sent by Margen67. Also, I'm not really sure if the saves are fixed in Master. In my case, Xenia Master still has broken saves.

**Comment by madmax2000** [comment]
> > In the config clear_memory_page_state = true
> > Or in the Xenia Manager Clear GPU Cache = true (is the same)
> > And there are no more visual glitches with the car models (no need for F5 anymore)
> > The game works perfect than!
> 
> Is this in Canary? Cause if it is, it needs to be in its separate Canary issue, the one above your message sent by Margen67. Also, I'm not really sure if the saves are fixed in Master. In my case, Xenia Master still has broken saves.
> 
> But to answer your question, its the first one. d3d12_clear_memory_page_state Set that to true and the glitched cars are gone.


Ok nice, i already see you posting it in the canvary thread. Yes i'm playing Canvary version.

Shouldn't be any performance drawback on moderate modern HW.

**Comment by Back2Life888** [comment]
> > > In the config clear_memory_page_state = true
> > > Or in the Xenia Manager Clear GPU Cache = true (is the same)
> > > And there are no more visual glitches with the car models (no need for F5 anymore)
> > > The game works perfect than!
> > 
> > 
> > Is this in Canary? Cause if it is, it needs to be in its separate Canary issue, the one above your message sent by Margen67. Also, I'm not really sure if the saves are fixed in Master. In my case, Xenia Master still has broken saves.
> > But to answer your question, its the first one. d3d12_clear_memory_page_state Set that to true and the glitched cars are gone.
> 
> Ok nice, i already see you posting it in the canvary thread. Yes i'm playing Canvary version.
> 
> Shouldn't be any performance drawback on moderate modern HW.
> 
> After that the game runs perfect with AMD GPUs.

RR6 in Xenia Master is still unplayable as far as I know. It still crashes and has no functional saves. This thread is for Xenia Master so Canary stuff here is kind of a no-no.

**Comment by Back2Life888** [comment]
> There are graphics bugs in Vulkan engine operation and Vsync off causes the game to speed up to 2x, etc

Vulkan is not being worked on at the moment. D3D12 is the main rendering engine that Xenia uses.
